scipy/optimize/tests/test_linprog.py

```diff
@@ -802,6 +802,44 @@ def test_bug_5400(self):
                     method=self.method, options=self.options)
             _assert_success(res, desired_fun=-106.63507541835018)
 
+    def test_issue_8174_stackoverflow(self):
+        # Test supplementary example from issue 8174.
+        # https://github.com/scipy/scipy/issues/8174
+        # https://stackoverflow.com/questions/47717012/linprog-in-scipy-optimize-checking-solution
+        c = np.array([1, 0, 0, 0, 0, 0, 0])
+        A_ub = -np.identity(7)
+        b_ub = np.array([[-2],[-2],[-2],[-2],[-2],[-2],[-2]])
+        A_eq = np.array([
+            [1, 1, 1, 1, 1, 1, 0],
+            [0.3, 1.3, 0.9, 0, 0, 0, -1],
+            [0.3, 0, 0, 0, 0, 0, -2/3],
+            [0, 0.65, 0, 0, 0, 0, -1/15],
+            [0, 0, 0.3, 0, 0, 0, -1/15]
+        ])
+        b_eq = np.array([[100],[0],[0],[0],[0]])
+
+        shoud_warn_A_not_full_rank = (
+            self.options.get('presolve', True)
+            and self.method == 'interior-point'
```

Comment:
Because only interior point currently implements presolve, this is redundant with the line above, right? And when simplex _does_ implement presolve, we would want it to be subject to the same condition. So I suggest removing this second condition.

Reply:
`pytest.warns(...)` fails the test if no warning is raised. Removing the second condition will cause the tests to fail *because* the simplex method does not implment presolve (yet).

Reply:
Maybe I'm wrong, but I thought currently simplex is never passed a 'presolve' option at all. So the first condition 'self.options.get('presolve', True)'  always be 'False' for simplex tests. Because 'and' is a short circuit operator, the second condition checking for interior point will never even be evaluated, so it is redundant. 'shoud_warn_A_not_full_rank' will be false, so the 'pytest.warns' part won't apply to simplex tests, so it wouldn't cause simplex tests to fail.

Please forgive me if I'm being dense! What part of this is wrong?

Reply:
Actually... I don't think `presolve` is ever going to be part of `self.options` for this test, even for interior point. So I think that `should_warn_A_not_full_rank` is _never_ `True`. Are you sure that the `with pytest.warns(OptimizeWarning)` is ever reached?

Since `presolve` is never in `self.options`, you don't really need to check it at all. Since `presolve` is on for IPM by default, I think what you want to do is something like:
if method = "interior point":
   run linprog and make warnings
else 
   run linprog normally
_assert_success part

Reply:
@mdhaber Yes you're right. I'll change this test.

Reply:
Yes you're right. I'll change the test. I wasn't sure if there are any tests for problems *without* presolve and wanted to be able to accomodate those. I'm pretty sure that, mathematically, every problem should return the same result regardless of whether presolve is used.